core/controllers/reader.py

```diff
@@ -1040,3 +1040,29 @@ def get(self):
             'question_dicts': question_dicts
         })
         self.render_json(self.values)
+
+
+class LearnerAnswerInfoSubmissionHandler(base.BaseHandler):
+    """Handles the learner answer details submission."""
+
+    @acl_decorators.can_play_exploration
+    def put(self, exploration_id):
+        """"Handles the PUT requests. Stores the answer details submitted
+        by the learner.
+        """
+
+        if not constants.ENABLE_SOLICIT_ANSWER_DETAILS_FEATURE:
+            raise self.PageNotFoundException
+
+        state_name = self.payload.get('state_name')
+        interaction_id = self.payload.get('interaction_id')
+        answer = self.payload.get('answer')
+        answer_details = self.payload.get('answer_details')
+        entity_type = feconf.ENTITY_TYPE_EXPLORATION
+        state_reference = (
+            stats_models.LearnerAnswerDetailsModel
+            .get_state_reference_for_exploration(exploration_id, state_name))
+        stats_services.record_learner_answer_info(
```

Comment:
is this method assures that there exists a given state_name and interaction_id for the corresponding exp_id before creating a new model?

Reply:
No such confirmation is done in the backend that the state name exists for the given exp, as from the frontend when the learner will submit the answer details it will fetch the state name and interaction from the frontend services and then pass it into it.

Reply:
I think we should add such validation otherwise we will be creating unwanted models, right? (/cc @aks681)

Reply:
Yeah, validation should be added for state names and exploration ids.

Reply:
Added

Reply:
Please add backend tests for the "invalid" cases.

Reply:
I have added cases, for invalid interaction and state.